include/f_core/util/linear_fit.hpp

```diff
@@ -0,0 +1,69 @@
+#include "f_core/util/circular_buffer.hpp"
+
+#include <array>
+#include <cstddef>
+
+template <typename T, std::size_t len> class RollingSum {
+  public:
+    static_assert(len > 0, "What is the sum of 0 elements? You probably don't want this (also it will break)");
+
+    using value_type = T;
+    static constexpr std::size_t size_ = len;
+
+    constexpr RollingSum(T start) : buf(start) { fill(start); }
+
+    constexpr std::size_t size() const { return size_; }
+
+    constexpr void fill(const value_type &start) {
+        buf.fill(start);
+        total = buf.oldest_sample();
+        for (std::size_t i = 1; i < buf.size(); i++) {
+            total = total + buf[1];
+        }
+    }
+
+    constexpr void feed(const value_type &new_value) {
+        value_type oldest = buf.oldest_sample();
+        total = total - oldest;
+        total = total + new_value;
+        buf.add_sample(new_value);
+    }
+    constexpr value_type sum() const { return total; }
+
+  private:
+    value_type total;
+    CircularBuffer<value_type, size_> buf;
+};
+
+template <typename T, std::size_t len> class MovingAvg {
```

Comment:
not actually used rn but useful for subsampling

Reply:
also was how grim detected boost (i was paranoid abt noise that was like 3 orders of magnitude lower than what i was worried abt)